text/trace/0174-http-semantic-conventions.md

```diff
@@ -0,0 +1,127 @@
+which will serve as a basis for [stabilizing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable)
+the [existing semantic conventions for HTTP](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md),
+which are currently in an [experimental](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#experimental)
+state. The goal is to declare HTTP semantic conventions stable before the
+end of 2021.
+
+## Motivation
+
+Most observability scenarios involve HTTP communication. For Distributed Tracing
+to be useful across the entire scenario, having good observability for
+HTTP is critical. To achieve this, OpenTelemetry must provide stable conventions
+and guidelines for instrumenting HTTP communication.
+
+Bringing the existing experimental semantic conventions for HTTP to a
+stable state is a crucial step for users and instrumentation authors, as it
+allows them to rely on [stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability),
+and thus to ship and use stable instrumentation.
+
+## Roadmap
+
+| Description | Done By     |
+|-------------|-------------|
+| This OTEP, consisting of scenarios/open questions and a proposed roadmap, is approved and merged. | 09/30/2021 |
+| [Stability guarantees](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#not-defined-semantic-conventions-stability) for semantic conventions are approved and merged. This is not strictly related to semantic conventions for HTTP but is a prerequisite for stabilizing any semantic conventions. | 09/30/2021 |
+| Separate PRs covering the scenarios and open questions in this document are approved and merged. | 10/29/2021 |
+| Proposed specification changes are verified by prototypes for the scenarios and examples below. | 11/15/2021 |
+| The [specification for HTTP semantic conventions for tracing](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md) is fully updated according to this OTEP and declared [stable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#stable). | 11/30/2021 |
+
+## Scope: scenarios and open questions
+
+> NOTE. The scope defined here is subject for discussions and can be adjusted.
+
+Scenarios and open questions mentioned below must be addressed via separate PRs.
+
+### Error status
+
+Per current spec 4xx must result in span with Error status. In many cases
+404/409 error criteria depends on the app though.
+
+### Required attribute sets
+
+> At least one of the following sets of attributes is required:
```

Comment:
Does this apply to metrics or just traces?

SHOULD we aim for unity between the two in attribute sets, mark this as explicitly out of scope?

Reply:
Just traces. Metrics are out of scope of this OTEP.